File: contest_and_submission_entities/trunk/src/java/main/com/topcoder/service/studio/contest/ChangeHistory.java
/*
 * Copyright (C) 2008 TopCoder Inc., All Rights Reserved.
 */
package com.topcoder.service.studio.contest;

import java.io.Serializable;
import java.util.Date;

/**
 * <p>
 * Represents the entity class contains information about changes made to a
 * contest, including the contest ID, the field changed, the username making the
 * change, a transaction ID that groups changes made at the same time, and the
 * old data and new data that represents the change.
 * </p>
 * 
 * <p>
 * Thread Safety: This entity is not thread safe since it is mutable.
 * </p>
 * 
 * @author Ghostar, superZZ
 * @version 1.0
 */
public class ChangeHistory implements Serializable {
    /**
     * 
     */
    private static final long serialVersionUID = 4782740922219724079L;
    
    /**
     * Represents the ContestId.
     */
    private Long contestId;

    /**
     * Updates the ContestId with the specified value.
     * 
     * @param contestId
     *            the contestId to set.
     */
    public void setContestId(Long contestId) {
        this.contestId = contestId;
    }

    /**
     * Returns the contestId.
     * 
     * @return the contestId.
     */
    public Long getContestId() {
        return contestId;
    }

    /**
     * Represents the TransactionId.
     */
    private Long transactionId;

    /**
     * Updates the TransactionId with the specified value.
     * 
     * @param transactionId
     *            the transactionId to set.
     */
    public void setTransactionId(Long transactionId) {
        this.transactionId = transactionId;
    }

    /**
     * Returns the transactionId.
     * 
     * @return the transactionId.
     */
    public Long getTransactionId() {
        return transactionId;
    }

    /**
     * Represents the Timestamp .
     */
    private Date timestamp;

    /**
     * Updates the Timestamp with the specified value.
     * 
     * @param timestamp
     *            the timestamp to set.
     */
    public void setTimestamp(Date timestamp) {
        this.timestamp = timestamp;
    }

    /**
     * Returns the timestamp.
     * 
     * @return the timestamp.
     */
    public Date getTimestamp() {
        return timestamp;
    }

    /**
     * Represents the UserName.
     */
    private String userName;

    /**
     * Updates the UserName with the specified value.
     * 
     * @param userName
     *            the userName to set.
     */
    public void setUserName(String userName) {
        this.userName = userName;
    }

    /**
     * Returns the userName.
     * 
     * @return the userName.
     */
    public String getUserName() {
        return userName;
    }

    /**
     * Represents the IsUserAdmin.
     */
    private boolean isUserAdmin;

    /**
     * Updates the IsUserAdmin with the specified value.
     * 
     * @param isUserAdmin
     *            the isUserAdmin to set.
     */
    public void setUserAdmin(boolean isUserAdmin) {
        this.isUserAdmin = isUserAdmin;
    }

    /**
     * Returns the isUserAdmin.
     * 
     * @return the isUserAdmin.
     */
    public boolean isUserAdmin() {
        return isUserAdmin;
    }

    /**
     * Represents the FieldName.
     */
    private String fieldName;

    /**
     * Updates the FieldName with the specified value.
     * 
     * @param fieldName
     *            the fieldName to set.
     */
    public void setFieldName(String fieldName) {
        this.fieldName = fieldName;
    }

    /**
     * Returns the fieldName.
     * 
     * @return the fieldName.
     */
    public String getFieldName() {
        return fieldName;
    }

    /**
     * Represents the OldData.
     */
    private String oldData;

    /**
     * Updates the OldData with the specified value.
     * 
     * @param oldData
     *            the oldData to set.
     */
    public void setOldData(String oldData) {
        this.oldData = oldData;
    }

    /**
     * Returns the oldData.
     * 
     * @return the oldData.
     */
    public String getOldData() {
        return oldData;
    }

    /**
     * Represents the NewData.
     */
    private String newData;

    /**
     * Updates the NewData with the specified value.
     * 
     * @param newData
     *            the newData to set.
     */
    public void setNewData(String newData) {
        this.newData = newData;
    }

    /**
     * Returns the newData.
     * 
     * @return the newData.
     */
    public String getNewData() {
        return newData;
    }
}

File: widget_bridge/trunk/src/js/main/js/topcoder/widgets/bridge/ChangeHistory.js
/**
 * Copyright (C) 2008 TopCoder Inc., All Rights Reserved.
 *
 * Widget Bridge 1.0
 */

/**
 * The constructor takes JSON object and assign value to each internal fields.
 * 
 * @param {JSONObject}
 *            json JSON object with will be converted to ChangeHistory
 * 
 * @consturctor
 * @namespace js.topcoder.widgets.bridge
 * @class ChangeHistory
 * 
 * This class is a client side mapping for ChangeHistory entity in service side.
 * It also encapsulates JSON object coversion. Thread Safety: This class is
 * mutable and not thread-safe. But JavaScript execution model is mostly
 * single-threaded. And there is no thread-synchronization features in
 * JavaScript, so it is not an issue here. Anyway, the external code should be
 * careful if several threads are used to communicate with this class.
 * 
 * @author TCSDEVELOPER
 * @version 1.0
 */
js.topcoder.widgets.bridge.ChangeHistory = function(json) {

	/**
	 * This variable represents ContestId. It can not be undefiend or null.
	 * 
	 * @private
	 * @type long
	 */
	var contestId;

	/**
	 * This variable represents TransactionId. It can not be undefiend or null.
	 * 
	 * @private
	 * @type long
	 */
	var transactionId;

	/**
	 * This variable represents UserName. It can not be undefiend or null.
	 * 
	 * @private
	 * @type string
	 */
	var userName;

	/**
	 * This variable represents timestamp. It can not be undefiend or null.
	 * 
	 * @private
	 * @type Date
	 */
	var timestamp;

	/**
	 * This variable represents FieldName. It can not be undefiend or null.
	 * 
	 * @private
	 * @type string
	 */
	var fieldName;

	/**
	 * This variable represents OldData. It can not be undefiend or null.
	 * 
	 * @private
	 * @type string
	 */
	var oldData;

	/**
	 * This variable represents IsUserAdmin. It can not be undefiend or null.
	 * 
	 * @private
	 * @type boolean
	 */
	var isUserAdmin;

	/**
	 * This variable represents NewData. It can not be undefiend or null.
	 * 
	 * @private
	 * @type string
	 */
	var newData;

	/**
	 * Abbreviation for Helper class.
	 */
	var Helper = js.topcoder.widgets.bridge.Helper;

	/**
	 * Abbreviation for IllegalArgumentException class.
	 */
	var IllegalArgumentException = js.topcoder.widgets.bridge.IllegalArgumentException;

	/**
	 * <p>
	 * Returns the ContestId.
	 * </p>
	 */
	this.getContestId = function /* long */() {
		return contestId;
	}

	/**
	 * <p>
	 * Sets the ContestId.
	 * </p>
	 */
	this.setContestId = function /* void */(/* long */contestId) {
		this.contestId = contestId;
	}


	/**
	 * <p>
	 * Returns the TransactionId.
	 * </p>
	 */
	this.getTransactionId = function /* long */() {
		return transactionId;
	}

	/**
	 * <p>
	 * Sets the TransactionId.
	 * </p>
	 */
	this.setTransactionId = function /* void */(/* long */transactionId) {
		this.transactionId = transactionId;
	}

	/**
	 * <p>
	 * Returns the timestamp.
	 * </p>
	 */
	this.getTimestamp = function /* Date */() {
		return timestamp;
	}

	/**
	 * <p>
	 * Sets the timestamp.
	 * </p>
	 */
	this.setTimestamp = function /* void */(/* Date */timestamp) {
		this.timestamp = timestamp;
	}

	/**
	 * <p>
	 * Returns the UserName.
	 * </p>
	 */
	this.getUserName = function /* string */() {
		return userName;
	}

	/**
	 * <p>
	 * Sets the UserName.
	 * </p>
	 */
	this.setUserName = function /* void */(/* string */userName) {
		this.userName = userName;
	}

	/**
	 * <p>
	 * Returns the FieldName.
	 * </p>
	 */
	this.getFieldName = function /* string */() {
		return fieldName;
	}

	/**
	 * <p>
	 * Sets the FieldName.
	 * </p>
	 */
	this.setFieldName = function /* void */(/* string */fieldName) {
		this.fieldName = fieldName;
	}

	/**
	 * <p>
	 * Returns the OldData.
	 * </p>
	 */
	this.getOldData = function /* string */() {
		return oldData;
	}

	/**
	 * <p>
	 * Sets the OldData.
	 * </p>
	 */
	this.setOldData = function /* void */(/* string */oldData) {
		this.oldData = oldData;
	}

	/**
	 * <p>
	 * Returns the NewData.
	 * </p>
	 */
	this.getNewData = function /* string */() {
		return newData;
	}

	/**
	 * <p>
	 * Sets the NewData.
	 * </p>
	 */
	this.setNewData = function /* void */(/* string */newData) {
		this.newData = newData;
	}

	/**
	 * <p>
	 * Returns the IsUserAdmin.
	 * </p>
	 */
	this.getIsUserAdmin = function /* boolean */() {
		return isUserAdmin;
	}

	/**
	 * <p>
	 * Sets the IsUserAdmin.
	 * </p>
	 */
	this.setIsUserAdmin = function /* void */(/* boolean */isUserAdmin) {
		this.isUserAdmin = isUserAdmin;
	}

	/**
	 * Convert the current instance into JSON string.
	 * 
	 * @return a JSON string representing current object
	 */
	this.toJSON = function() {
		var result = "{";

		result += "\"contestId\" : " + this.contestId + ", ";
		result += "\"transactionId\" : " + this.transactionId + ", ";
		result += "\"timestamp\": \"" + this.timestamp + "\", ";
		result += "\"userName\": \"" + this.userName + "\"" + ", ";
		result += "\"fieldName\": \"" + this.fieldName + "\"" + ", ";
		result += "\"oldData\": \"" + this.oldData + "\"" + ", ";
		result += "\"newData\": \"" + this.newData + "\"" + ", ";
		result += "\"isUserAdmin\": " + this.isUserAdmin;

		result += "}";
		return result;
	}

	
	// constructor code
	if (json != null) {
		this.setContestId(json.contestId);
		this.setTransactionId(json.transactionId);
		this.setTimestamp(json.timestamp);
		this.setUserName(json.userName);
		this.setFieldName(json.fieldName);
		this.setOldData(json.oldData);
		this.setNewData(json.newData);
		this.setIsUserAdmin(json.isUserAdmin);
	}
}